src/API/Google/AdsAsset.php

```diff
@@ -18,7 +26,97 @@
  *
  * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Google
  */
-class AdsAsset {
+class AdsAsset implements OptionsAwareInterface {
+
+	use OptionsAwareTrait;
+
+	/**
+	 * Temporary ID to use within a batch job.
+	 * A negative number which is unique for all the created resources.
+	 *
+	 * @var int
+	 */
+	protected const TEMPORARY_ID = -5;
+
+	/**
+	 * Return a temporary resource name for the asset.
+	 *
+	 * @param int $temporary_id The temporary ID to use for the asset.
+	 *
+	 * @return string The Asset resource name.
+	 */
+	protected function temporary_resource_name( $temporary_id = self::TEMPORARY_ID ): string {
+		return ResourceNames::forAsset( $this->options->get_ads_id(), $temporary_id );
+	}
+
+	/**
+	 * Returns the asset type for the given field type.
+	 *
+	 * @param string $field_type The field type.
+	 *
+	 * @return int The asset type.
+	 * @throws Exception If the field type is not supported.
+	 */
+	protected function get_asset_type_by_field_type( string $field_type ): int {
+		switch ( $field_type ) {
+			case AssetFieldType::LOGO:
+			case AssetFieldType::MARKETING_IMAGE:
+			case AssetFieldType::SQUARE_MARKETING_IMAGE:
+				return AssetType::IMAGE;
+			case AssetFieldType::CALL_TO_ACTION_SELECTION:
+				return AssetType::CALL_TO_ACTION;
+			case AssetFieldType::HEADLINE:
+			case AssetFieldType::LONG_HEADLINE:
+			case AssetFieldType::DESCRIPTION:
+			case AssetFieldType::BUSINESS_NAME:
+				return AssetType::TEXT;
+			default:
+				throw new Exception( 'Asset Field type not supported' );
+		}
+
+	}
+
+	/**
+	 * Returns an operation to create a text asset.
+	 *
+	 * @param array $data The asset data.
+	 * @param int   $temporary_id The temporary ID to use for the asset.
+	 *
+	 * @return MutateOperation The create asset operation.
+	 * @throws Exception If the asset type is not supported or if the image url is not a valid url.
+	 */
+	public function create_operation( array $data, int $temporary_id = self::TEMPORARY_ID ): MutateOperation {
+		$asset = new Asset(
+			[
+				'resource_name' => $this->temporary_resource_name( $temporary_id ),
+			]
+		);
+
+		switch ( $this->get_asset_type_by_field_type( $data['field_type'] ) ) {
+			case AssetType::CALL_TO_ACTION:
+				$asset->setCallToActionAsset( new CallToActionAsset( [ 'call_to_action' => CallToActionType::number( $data['content'] ) ] ) );
+				break;
+			case AssetType::IMAGE:
+				$image_data = wp_remote_get( $data['content'] );
+
+				if ( is_wp_error( $image_data ) || empty( $image_data['body'] ) ) {
+					throw new Exception( 'Incorrect image asset url.' );
+				}
```

Comment:
❓  Wondering if we can wrap this in the WP Proxy

Reply:
I have added `wp_remote_get` to the WP proxy as it makes easier the testing for the Image assets, however, I am not sure if we should include throwing the `exception`  in the WP Proxy as it will depend on the consumer how it wants to treat the case where exists a `wp_error` or if we want to consider an empty body as an `Exception`. See [b7664f0](https://github.com/woocommerce/google-listings-and-ads/pull/1832/commits/b7664f060f631c6f110959f10667e94b7ad314f1)